binding.cc

```diff
@@ -444,6 +444,14 @@ struct BaseWorker {
       argv = CreateCodeError(env, "LEVEL_NOT_FOUND", errMsg_);
     } else if (status_.IsCorruption()) {
       argv = CreateCodeError(env, "LEVEL_CORRUPTION", errMsg_);
+    } else if (status_.IsIOError()) {
+      if (strlen(errMsg_) > 15 && strncmp("IO error: lock ", errMsg_, 15) == 0) { // env_posix.cc
+        argv = CreateCodeError(env, "LEVEL_LOCKED", errMsg_);
+      } else if (strlen(errMsg_) > 19 && strncmp("IO error: LockFile ", errMsg_, 19) == 0) { // env_win.cc
+        argv = CreateCodeError(env, "LEVEL_LOCKED", errMsg_);
```

Comment:
This platform difference will be gone once we upgrade LevelDB - but we'll still have to compare the error message. I don't see a cleaner way atm, short of patching upstream to return a more specific `leveldb::Status`.